### PR TITLE
HPCC-15545 ESDL Tool and WsESDLConfig to make ESDL Serv name optional

### PR DESCRIPTION
**Pull request by rpastrana**
- ESDL Service name is now optional when publishing an ESDL definition
  If defintion only contains one service definition

- ESDL Service name is now optional when publishing an ESDL Binding
  If defintion only contains one service definition- Removes extra space in placeholder ip list text

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [x] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@wangkx please review

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp [hunk 1]
```diff
@@ -300,25 +300,29 @@ bool CWsESDLConfigEx::existsESDLMethodDef(const char * esdlDefinitionName, unsig
...
+                if ((servicesCount == 1 && !esdlServiceName.length()) || stricmp(pCurrService->queryProp("@name"), esdlServiceName.str())==0)
                 {
                     Owned<IPropertyTreeIterator> it2 = esxdl->getElements("EsdlMethod");
                     ForEach(*it2)
                     {
                         IPropertyTree* pChildNode = &it2->query();
                         if (stricmp(pChildNode->queryProp("@name"), methodName)==0)
                         {
+                            if (!esdlServiceName.length())
+                                esdlServiceName.set(pCurrService->queryProp("@name"));
```
Comment (wangkx) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: What if esdlServiceName has some value which != pCurrService->queryProp("@name")?

File: tools/esdlcmd/esdl-publish.cpp [hunk 2]
```diff
@@ -236,8 +245,8 @@ class EsdlPublishCmd : public EsdlPublishCmdCommon
         if (optSource.isEmpty())
             throw MakeStringException( 0, "Source ESDL definition file (ecm|esdl|xml) must be provided" );
 
-        if (optESDLService.isEmpty())
-            throw MakeStringException( 0, "Name of ESDL based service must be provided" );
+        //if (optESDLService.isEmpty())
+        //    throw MakeStringException( 0, "Name of ESDL based service must be provided" );
```
Comment (wangkx) [inline tools/esdlcmd/esdl-publish.cpp]: Remove?

**Review by wangkx — COMMENTED** [review]
@rpastrana 2 questions

File: tools/esdlcmd/esdl-publish.cpp (hunk 2 above)
Comment (rpastrana) [inline tools/esdlcmd/esdl-publish.cpp]: Will remove

File: esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp (hunk 1 above)
Comment (rpastrana) [inline esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp]: I think the conditional at 309 prevents that scenario; we only enter this block if either esdlServiceName == @name, or esdlService is empty and there's only one service...

**Comment by rpastrana** [comment]
@wangkx changes squashed, please review.

**Comment by wangkx** [comment]
@rpastrana looks fine.

**Comment by rpastrana** [comment]
@richardkchapman please merge

**Pull request merged**